updatesh_d/scriptutil.d

```diff
@@ -0,0 +1,184 @@
+module scriptutil;
+import core.sys.posix.unistd:geteuid;
+static import std.ascii;
+import std.typecons:Tuple,tuple;
+import std.datetime:Clock,SysTime;
+import std.digest.digest:digest;
+import std.digest.crc:crcHexString,CRC32;
+import std.file:mkdir,exists,isDir,File,dirSeparator;
+import std.process:executeShell;
+import std.string:strip,indexOf,munch;
+import std.array:join;
+import std.math:pow;
+import std.algorithm:max;
+import std.random:uniform;
+import std.conv:to;
+import std.stdio:writefln,stdout;
+// shamelessly nicked from http://www.mktemp.org/repos/mktemp/file/e4089ef40fd0/mkdtemp.c
+
+
+private string makeHash(immutable string path)
```

Comment:
`immutable` is a bit redundant here and would be appropriate mostly for large functions, just use `in` or nothing